### PR TITLE
add M5 CoreInk project

### PR DESCRIPTION
**Pull request by xas**
## Description

* add CoreInk project for the [CoreInk](https://docs.m5stack.com/en/core/coreink) product

## Motivation and Context

* add ready-to-go properties for the CoreInk product
* 
## How Has This Been Tested?

see README file for implemented properties and testing result

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

## Summary by CodeRabbit

- **New Features**
	- Introduced support for the `nanoFramework.CoreInk` package in the pipeline.
	- Added a comprehensive `packages.lock.json` file for `nanoFramework.CoreInk`, detailing essential project dependencies, including various nanoFramework libraries.

- **Bug Fixes**
	- Ensured integrity verification through content hashes in the new dependencies manifest.

### CODE REVIEW

**Comment by xas** [comment]
Draft mode as missing for:

* [ ] nuGet generation implementation
* [ ] [PR validation](https://github.com/nanoframework/nanoFramework.IoT.Device/pull/902) for screen implementation
* [ ] screen implementation

File: nanoFramework.CoreInk/Properties/AssemblyInfo.cs
Comment (Ellerbach) [inline nanoFramework.CoreInk/Properties/AssemblyInfo.cs:19]: You don't need a specific native version I think

File: nanoFramework.CoreInk/README.md
Comment (Ellerbach) [inline nanoFramework.CoreInk/README.md:1]: I guess that your goal to create this specific README is to use it for the nuget. In that case, looks good. Otherwise, add the key elements into the main readme.

File: nanoFramework.CoreInk/M5CoreInk.cs
Comment (Ellerbach) [inline nanoFramework.CoreInk/M5CoreInk.cs:148]: will you add the screen once it will be merged from the IoT Device repo?

File: nanoFramework.CoreInk/packages.config
Comment (Ellerbach) [inline nanoFramework.CoreInk/packages.config:1]: you also need a lock package
And you'll also need a specific nuspec in the root folder (see the other ones as example)
And most likely adjust the azure pipeline to add the build and package for this one!

File: nanoFramework.CoreInk/M5CoreInk.cs
Comment (xas) [reply nanoFramework.CoreInk/M5CoreInk.cs:148]: That's was the idea.

It is also possible to do a v1 without screen and later a v2 with it. But not sure if pertinent..

File: nanoFramework.CoreInk/README.md
Comment (xas) [reply nanoFramework.CoreInk/README.md:1]: I didn't think about including it in the nuget. It was more to display a current state of code , and not render the main one too full of data (we can create a README per core also with the same kind of information)

File: nanoFramework.CoreInk/Properties/AssemblyInfo.cs
Comment (xas) [reply nanoFramework.CoreInk/Properties/AssemblyInfo.cs:19]: Guilty of copy/paste... 😅 

I will cleanup

File: nanoFramework.CoreInk/M5CoreInk.cs
Comment (Ellerbach) [reply nanoFramework.CoreInk/M5CoreInk.cs:148]: There is no urgency, let's merge properly the screen and you'll be able to add it here. So it will be a nice version.

**Comment by networkfusion** [comment]
@xas any update on this?

**Comment by xas** [comment]
@networkfusion Yeah sorry, I was waiting for the merge of the E-Ink driver and then _lot of work_ happened 😞 

I will try to update the PR this week

File: nanoFramework.CoreInk/Properties/AssemblyInfo.cs
Comment (Ellerbach) [reply nanoFramework.CoreInk/Properties/AssemblyInfo.cs:19]: You still need to clean up this part ;-)

**Review by Ellerbach — COMMENTED** [review]
Looks great, thanks a lot for moving it forward. Just a little thing and you'll be good to go!

**Comment by josesimoes** [comment]
@xas hey hey! 😃 
Are you planning on resuming work on this PR?

**Comment by xas** [comment]
> @xas hey hey! 😃 Are you planning on resuming work on this PR?

yeah, sorry, I wish I had more time. Plus updating packages/nuspec require some attention 😭

**Comment by josesimoes** [comment]
@xas glad to see that you're able to finish this. 👍🏻 
Let me know if you need any help with anything that is blocking you.

**Comment by Ellerbach** [comment]
Note: we can ignore the line duplication. It's mainly "normal" regarding how we've been setting up the projects.

File: azure-pipelines.yml
Comment (Ellerbach) [inline azure-pipelines.yml:96]:
```suggestion

```

File: nanoFramework.CoreInk/M5CoreInk.cs
Comment (Ellerbach) [inline nanoFramework.CoreInk/M5CoreInk.cs:167]: is there any way to skip initializing the screen if it's been already initialized?

File: nanoFramework.CoreInk/M5CoreInk.cs
Comment (Ellerbach) [inline nanoFramework.CoreInk/M5CoreInk.cs:18]:
```suggestion
    /// M5 CoreInk board.
```

File: nanoFramework.CoreInk/M5CoreInk.cs
Comment (Ellerbach) [inline nanoFramework.CoreInk/M5CoreInk.cs:189]:
```suggestion
        /// Gets an ADC channel.
```

File: nanoFramework.CoreInk/M5CoreInk.cs
Comment (Ellerbach) [inline nanoFramework.CoreInk/M5CoreInk.cs:191]:
```suggestion
        /// <param name="gpioNumber">The GPIO pin number.</param>
```

File: nanoFramework.CoreInk/M5CoreInk.cs
Comment (Ellerbach) [reply nanoFramework.CoreInk/M5CoreInk.cs:167]: ok, forgot about it, just checked the function and it doesn't initialized it if not needed.

File: nanoFramework.CoreInk/README.md
Comment (Ellerbach) [inline nanoFramework.CoreInk/README.md:39]:
```suggestion
```text
```

**Review by Ellerbach — APPROVED** [review]
Minimal comments and suggestions I'll merge to speed up the process :-)

**Pull request merged**
